### ISSUE TITLE
Can not create multiple queries with the same name

### ISSUE THREAD

**Issue by stuarthendren** [leangen/graphql-spqr#35]
When I try to use the same query name, with different arguments, I get a an exception

```
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [graphql.schema.GraphQLSchema]: Factory method 'schema' threw exception; nested exception is java.lang.IllegalStateException: Not all resolvers expect the same source types
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189) ~[spring-beans-4.3.11.RELEASE.jar:4.3.11.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588) ~[spring-beans-4.3.11.RELEASE.jar:4.3.11.RELEASE]
	... 67 common frames omitted
Caused by: java.lang.IllegalStateException: Not all resolvers expect the same source types
	at io.leangen.graphql.metadata.strategy.query.DefaultOperationBuilder.resolveContextTypes(DefaultOperationBuilder.java:77) ~[spqr-0.9.3.jar:na]
	at io.leangen.graphql.metadata.strategy.query.DefaultOperationBuilder.buildQuery(DefaultOperationBuilder.java:34) ~[spqr-0.9.3.jar:na]
	at io.leangen.graphql.generator.OperationRepository.lambda$buildQueries$39(OperationRepository.java:38) ~[spqr-0.9.3.jar:na]
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[na:1.8.0_112]
	at java.util.HashMap$EntrySpliterator.forEachRemaining(HashMap.java:1691) ~[na:1.8.0_112]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481) ~[na:1.8.0_112]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471) ~[na:1.8.0_112]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[na:1.8.0_112]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:1.8.0_112]
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499) ~[na:1.8.0_112]
	at io.leangen.graphql.generator.OperationRepository.buildQueries(OperationRepository.java:39) ~[spqr-0.9.3.jar:na]
	at io.leangen.graphql.generator.OperationRepository.<init>(OperationRepository.java:31) ~[spqr-0.9.3.jar:na]
	at io.leangen.graphql.GraphQLSchemaGenerator.generate(GraphQLSchemaGenerator.java:685) ~[spqr-0.9.3.jar:na]
	at io.committed.BaleenGrahpQl.schema(BaleenGrahpQl.java:43) [classes/:na]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_112]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_112]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_112]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_112]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162) ~[spring-beans-4.3.11.RELEASE.jar:4.3.11.RELEASE]
	... 68 common frames omitted
```

**Comment by stuarthendren** [comment]
It looks like this is when mixing context and argument based queries, for example:

```
  @GraphQLQuery(name = "relations")
  public List<Relation> getByDocument(@GraphQLArgument(name = "documentId") @GraphQLId String id) {
    return relations.findByDocId(id);
  }

  @GraphQLQuery(name = "relations")
  public List<Relation> getRelations(@GraphQLContext Document document) {
    return getByDocument(document.getId());
  }
```

maybe this is not allowed?

**Comment by kaqqao** [comment]
You are right. Currently, if one of the resolver methods has a `@GraphQLContext`  attribute, all the resolvers must have the same.

This is actually a needless and unintuitive limitation, and I have already made the changes to fix it, but haven't yet pushed them. I'll resume the work in about 10 days.

**Comment by freben** [comment]
@kaqqao Any news on this? I am being bitten by this too, but in a slightly different way:

```
  @GraphQLQuery(name = "children")
  public List<A> childrenOfA(@GraphQLContext A a) {
    return singletonList(new A(2));
  }

  @GraphQLQuery(name = "children")
  public List<B> childrenOfB(@GraphQLContext B b) {
    return singletonList(new B(2));
  }
```

Even having the same name on a field under a context fails with the same error.

**Comment by kaqqao** [comment]
Fixed in master now :) Please verify it works for you if you can. If not, I just need to fix one more issue and I'll release 0.9.4. After that I'll start the work on moving SPQR to the latest graphql-java.

**Comment by kaqqao** [comment]
I'm aiming to release v0.9.4 (containing this fix) tomorrow.
